extensions/amp-viewer-integration/0.1/amp-viewer-integration.js

```diff
@@ -112,11 +116,13 @@ export class AmpViewerIntegration {
             e.source === source &&
             data.app == APP &&
             data.name == 'handshake-poll') {
-          if (!e.ports || !e.ports.length) {
+          if (this.isWebView_ && (!e.ports || !e.ports.length)) {
             throw new Error(
               'Did not receive communication port from the Viewer!');
           }
-          resolve(e.ports[0]);
+          const port = this.isWebView_ ? e.ports[0] : new WindowPortEmulator(
```

Comment:
Let's change this to
```
const port = e.ports && e.ports.length > 0 ? e.ports[0] : new WindowPortEmu(...);
```

This way, we'll allow port use for both webviews and iframes and it will be up to the parent window to decide if it wants to use port or not.